File: day1/day1.ts
import fs from "fs";

// read file input.txt
const input = fs.readFileSync("input.txt", "utf8");

const measurements : number[] = input.split("\n").map(s => parseInt(s));

let count : number = 0;
for (let i = 0; i < measurements.length - 3; i++) {
    let sum1 = measurements[i] + measurements[i+1] + measurements[i+2];
    let sum2 = measurements[i+1] + measurements[i+2] + measurements[i+3];
    if (sum2 > sum1) {
        count++;
    }
}

console.log(count);

File: day1/day1.kt
import java.io.File

fun main() {
    val input = File("input.txt").readText(Charsets.UTF_8).split("\n").map { it.toInt() };
    // Part one
    println(input.windowed(2).count { (a,b) -> a < b });
    // Part two
    println(input.windowed(3).windowed(2).count { (a,b) -> a.sum() < b.sum()});
}